scipy/stats/_distn_infrastructure.py

```diff
@@ -1634,7 +1635,28 @@ def _penalized_nnlf(self, theta, x):
             return inf
         x = asarray((x-loc) / scale)
         n_log_scale = len(x) * log(scale)
-        return self._nnlf_and_penalty(x, args) + n_log_scale
+        return self._nlff_and_penalty(x, args, self._logpxf) + n_log_scale
+
+    def _penalized_nlpsf(self, theta, x):
+        """Penalized negative log product spacing function.
+        i.e., - sum (log (diff (cdf (x, theta))), axis=0) + penalty
+        where theta are the parameters (including loc and scale)
+        Follows [1] of scipy.stats.fit
+        """
+        loc, scale, args = self._unpack_loc_scale(theta)
+        if not self._argcheck(*args) or scale <= 0:
+            return inf
+        x = (np.sort(x) - loc)/scale
+
+        def log_psf(x, *args):
+            # simplest possible implementation for starters
+            x, lj = np.unique(x, return_counts=True)
+            cdf_data = self._cdf(x, *args) if x.size else []
+            cdf = np.concatenate(([0], cdf_data))
```

Comment:
I misread the paper before. If `cdf_data` does not contain 1, it should be appended.